### PR TITLE
Readcount annotation fix

### PR DESCRIPTION
**Pull request by ernfrid**
Indel alleles are not being comprehensively handled in variant reporting currently. This is a work in progress to try to remedy that. I am still running some live tests on the new code to verify we've addressed this everywhere, but would greatly appreciate some review.

### CODE REVIEW

File: lib/perl/Genome/File/Vcf/BamReadcountUtilities.t [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env perl
```
Comment (tmooney) [inline lib/perl/Genome/File/Vcf/BamReadcountUtilities.t]: We tend to use `#!/usr/bin/env genome-perl` in our tests.

**Comment by ernfrid** [comment]
Fixed. Thanks Tom.

**Comment by ernfrid** [comment]
I just added a bit of additional code to address some bam-readcount bugs. I will need to re-run my out-of-band testing, but I think this is ready for any additional review/feedback.

**Comment by ernfrid** [comment]
The VAF code is not going to work due to upstream complexity in how bam-readcount is run. I've reverted.

**Comment by ernfrid** [comment]
I think this is ready for serious review now.

**Comment by mkiwala** [comment]
+1 :koala:

**Comment by apregier** [comment]
+1

**Pull request merged**